### PR TITLE
Run process monitor in root network namespace

### PR DESCRIPTION
**Pull request by Yumasi**
### What does this PR do?

This PR runs the netlink based process monitor in the root network namespace. Netlink is affected by NetNS, and in case system-probe is running in a containerised environment, it would only see the events of the container, instead of the host machine.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by nplanel — APPROVED** [review]

**Review by guyarb — CHANGES_REQUESTED** [review]
Such a PR cannot be merged without proper UTs or QA steps

File: pkg/process/monitor/process_monitor_test.go [hunk 1]
```diff
@@ -174,3 +177,29 @@ func TestProcessMonitorRefcount(t *testing.T) {
...
+	pm := GetProcessMonitor()
+	unsubscribeExec, err := pm.Subscribe(&ProcessCallback{
+		Event:    EXEC,
+		Metadata: ANY,
+		Callback: func(pid uint32) {
+			execSet.Store(pid, struct{}{})
+		},
+	})
+	defer unsubscribeExec()
+	require.NoError(t, err, "could not subscribe to EXEC events")
```
Comment (guyarb) [inline pkg/process/monitor/process_monitor_test.go:193]: on error, `unsubscribeExec` is nil
```suggestion
	require.NoError(t, err, "could not subscribe to EXEC events")
	defer unsubscribeExec()
```

File: pkg/process/monitor/process_monitor_test.go [hunk 2]
```diff
@@ -174,3 +177,29 @@ func TestProcessMonitorRefcount(t *testing.T) {
...
+	monNs, err := netns.New()
+	require.NoError(t, err, "could not create network namespace for process monitor")
+
+	require.NoError(t, procutils.WithNS(monNs, pm.Initialize), "could not start process monitor in netNS")
+
+	cmd := exec.Command("/bin/echo")
+	require.NoError(t, cmd.Run(), "could not run process in root namespace")
+
+	_, captured := execSet.Load(uint32(cmd.ProcessState.Pid()))
+	require.True(t, captured, "did not capture process EXEC from root namespace")
```
Comment (guyarb) [inline pkg/process/monitor/process_monitor_test.go:204]: wrap it with `require.Eventually` to ensure the test is not flaky due to raced

**Review by guyarb — CHANGES_REQUESTED** [review]

**Comment by guyarb** [comment]
did you deploy it on k8s and verified it works correctly?

**Comment by Yumasi** [comment]
> did you deploy it on k8s and verified it works correctly?

Yes, it's been used in the load testing environment as a fix for hooking Java processes.

**Review by guyarb — APPROVED** [review]

**Review by xornivore — APPROVED** [review]

**Pull request merged**